src/runtime/interface-info.ts

```diff
@@ -293,19 +296,19 @@ ${this._slotsToManifestString()}
     return copy;
   }
 
-  static _updateTypeVar(typeVar, update) {
+  static _updateTypeVar(typeVar, update): void {
     typeVar.object[typeVar.field] = update(typeVar.object[typeVar.field]);
   }
 
-  static isTypeVar(reference) {
+  static isTypeVar(reference): boolean {
     return (reference instanceof Type) && reference.hasProperty(r => r instanceof TypeVariable);
   }
 
-  static mustMatch(reference) {
+  static mustMatch(reference): boolean {
     return !(reference == undefined || InterfaceInfo.isTypeVar(reference));
   }
 
-  static handlesMatch(interfaceHandle, particleHandle) {
+  static handlesMatch(interfaceHandle, particleHandle): boolean|[{var, value, direction}] {
```

Comment:
is this the right return value? It implies that the list always has exactly one member.